File: POS System/POS System/SysUser.vb
Imports System.Data.SqlClient

Public Class SysUser
    Dim Pos As Integer
    Dim EditString As String
    Dim objCellStyle As New DataGridViewCellStyle()
    Dim Addition, Editing, RollBack As Boolean

    Private Sub frmSysUser_KeyUp(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles Me.KeyUp
        ' If user pressed the Enter key
        If e.KeyCode = Keys.Enter And Not (e.Alt Or e.Control) Then
            e.SuppressKeyPress = True
            ' Make sure that the active control is a TextBox control
            ' Do not use the Enter key as tab when a button has the focus!
            If Me.ActiveControl.GetType Is GetType(TextBox) Or _
                    Me.ActiveControl.GetType Is GetType(ComboBox) Or _
                    Me.ActiveControl.GetType Is GetType(DateTimePicker) Then
                ' Use Shift + Enter to move backwords through the tab order
                If e.Shift Then
                    Me.ProcessTabKey(False)
                Else
                    Me.ProcessTabKey(True)
                End If
            End If
        End If
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        If gblPrliviges > 2 Then
            MsgBox("You are not authorized, illegal operating.", MsgBoxStyle.Information, My.Application.Info.ProductName)
            Me.Close()
            Exit Sub
        End If
        Call GetConnectionString()
        Call LoadDB()

    End Sub

    Private Sub LoadDB()
        Try
            Dim CommandText As String = "SELECT * FROM USERS ORDER BY USERID"
            ExecuteDatasetQuery(SqlDataAdapter, SqlDataSet, CommandText, "USERS", Connection)
            SqlDataAdapter.Fill(Me.UserDataSet, "USERS")
            Me.UserbindingSource.DataSource = Me.UserDataSet
            Me.UserbindingSource.DataMember = "USERS"
            Me.txtUserID.DataBindings.Add("TEXT", UserbindingSource, "USERID")
            Me.txtUserName.DataBindings.Add("TEXT", Me.UserbindingSource, "USERNAME")
            Me.txtName.DataBindings.Add("TEXT", Me.UserbindingSource, "NAME")
            Me.cmbUserType.DataBindings.Add("TEXT", UserbindingSource, "USERROLE")
            Me.txtLevel.DataBindings.Add("TEXT", UserbindingSource, "SECURITYLEVEL")
            Me.txtPwd.DataBindings.Add("TEXT", UserbindingSource, "PASSWORD")
            Me.txtCNIC.DataBindings.Add("TEXT", Me.UserbindingSource, "CARDNUMBER")
            Addition = False
            Editing = False
            Me.ToolStripSaveButton.Enabled = False
            Me.Panel1.Enabled = False
            objCellStyle.BackColor = Color.WhiteSmoke
            Me.UsersDataGridView.AlternatingRowsDefaultCellStyle = objCellStyle
        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Information, My.Application.Info.ProductName)
            Me.Close()
        End Try
    End Sub

    Private Sub BindingNavigatorAddNewItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorAddNewItem.Click
        Panel1.Enabled = True
        Addition = True
        RollBack = True
        Editing = False
        Me.ToolStripSaveButton.Enabled = True
        Me.ToolStripCloseButton.Text = "&Cancel"
        Me.txtUserName.Focus()
    End Sub

    Private Sub ToolStripEditButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripEditButton.Click

        EditString = txtUserID.Text.Trim.ToUpper
        Editing = True
        Addition = False
        Panel1.Enabled = True
        Me.ToolStripSaveButton.Enabled = True
        Me.ToolStripCloseButton.Text = "&Cancel"
        EditString = Me.txtUserID.Text.Trim
        Me.txtUserName.Focus()

    End Sub

    Private Sub ToolStripCloseButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripCloseButton.Click
        If Me.ToolStripCloseButton.Text = "&Close" Then
            'myConn.Close()
            Me.UserbindingSource.Dispose()
            Me.Close()
        Else
            Me.ToolStripCloseButton.Text = "&Close"
            Me.ToolStripSaveButton.Enabled = False
            Panel1.Enabled = False
            Call RefreshDB()
            Addition = False
            Editing = False
        End If
    End Sub

    Private Sub RefreshDB()
        Try
            Me.UserDataSet.Clear()
            Dim CommandText As String = "SELECT * FROM USERS ORDER BY USERID"
            ExecuteDatasetQuery(SqlDataAdapter, SqlDataSet, CommandText, "USERS", Connection)
            SqlDataAdapter.Fill(Me.UserDataSet, "USERS")
            Me.UserbindingSource.DataSource = Me.UserDataSet
            Me.UserbindingSource.DataMember = "USERS"
            Me.UsersDataGridView.RefreshEdit()
        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Information, My.Application.Info.ProductName)
            Me.Close()
        End Try
    End Sub

    Private Sub BindingNavigatorDeleteItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorDeleteItem.Click

        Dim DeleteQuery As String = ""
        If Me.ToolStripCloseButton.Text = "&Cancel" Then Me.ToolStripCloseButton.Text = "&Close"
        If MsgBox("Are you sure to delete ('" & txtUserName.Text.Trim.ToUpper & "') ?", MsgBoxStyle.Information + MsgBoxStyle.OkCancel, My.Application.Info.ProductName) = MsgBoxResult.Ok Then
            DeleteQuery = "DELETE FROM USERS WHERE USERNAME = '" & txtUserName.Text.Trim & "'"
            ExecuteDatasetQuery(SqlDataAdapter, SqlDataSet, DeleteQuery, "USERS", Connection)
        Else
            Me.UsersDataGridView.RefreshEdit()
            Me.UserBindingNavigator.Refresh()
            Call ToolStripCloseButton_Click(sender, e)
            Exit Sub
        End If
    End Sub

    Private Sub ToolStripSaveButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripSaveButton.Click
        Dim MaxID As String = GenerateMax_ID("USERS", "USERID")
        Dim Edit_Update_Query As String = ""
        If Editing = True And Addition = False Then Edit_Update_Query = "UPDATE USERS SET USERNAME = '" & txtUserName.Text.Trim.ToLower & "', PASSWORD = '" & txtPwd.Text.Trim & "', USERROLE = '" & StrConv(cmbUserType.Text.Trim, VbStrConv.ProperCase) & "', SECURITYLEVEL = '" & txtLevel.Text.Trim & "', NAME = '" & StrConv(txtName.Text.Trim, VbStrConv.ProperCase) & "'  WHERE USERID = '" & Val(EditString) & "'"
        If Addition = True And Editing = False Then Edit_Update_Query = "INSERT INTO USERS VALUES('" & MaxID & "','" & txtUserName.Text.Trim.ToLower & "','" & txtPwd.Text.Trim & "','" & StrConv(txtName.Text.Trim, VbStrConv.ProperCase) & "','" & StrConv(cmbUserType.Text.Trim, VbStrConv.ProperCase) & "','" & Val(txtLevel.Text.Trim.ToUpper) & "',0,'" & txtCNIC.Text.Trim & "')"
        ExecuteAdapterQuery(SqlDataAdapter, Edit_Update_Query, Connection)
        Call RefreshDB()
        ToolStripSaveButton.Enabled = False
        Panel1.Enabled = False
        Me.ToolStripCloseButton.Text = "&Close"
        Connection.Close()
    End Sub

    Private Sub txtName_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtName.KeyPress
        Call ValidateAlphaKeyPress(sender, e, txtName)
    End Sub

    Private Sub txtUserName_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtUserName.KeyPress
        If e.KeyChar = Chr(8) Then e.Handled = False
    End Sub

    Private Sub txtPwd_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtPwd.KeyPress
        If e.KeyChar = Chr(8) Then e.Handled = False
    End Sub

    Private Sub cmbUserType_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles cmbUserType.KeyPress
        If Not (Char.IsLetter(e.KeyChar) Or Char.IsControl(e.KeyChar)) Then e.Handled = True
    End Sub

    Private Sub cmbUserType_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmbUserType.SelectedIndexChanged

        If cmbUserType.Text = "Admin" Then
            txtLevel.Text = 1
        ElseIf cmbUserType.Text = "Super User" Then
            txtLevel.Text = 2
        ElseIf cmbUserType.Text = "User" Then
            txtLevel.Text = 3
        End If

    End Sub

End Class

File: POS System - Retaurant - 2022/POS System/SysUser.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace POS_System
{

    public partial class SysUser
    {
        private int Pos;
        private string EditString;
        private DataGridViewCellStyle objCellStyle = new DataGridViewCellStyle();
        private bool Addition, Editing, RollBack;

        public SysUser()
        {
            InitializeComponent();
        }

        private void frmSysUser_KeyUp(object sender, KeyEventArgs e)
        {
            // If user pressed the Enter key
            if (e.KeyCode == Keys.Enter & !(e.Alt | e.Control))
            {
                e.SuppressKeyPress = true;
                // Make sure that the active control is a TextBox control
                // Do not use the Enter key as tab when a button has the focus!
                if (ReferenceEquals(ActiveControl.GetType(), typeof(TextBox)) | ReferenceEquals(ActiveControl.GetType(), typeof(ComboBox)) | ReferenceEquals(ActiveControl.GetType(), typeof(DateTimePicker)))

                {
                    // Use Shift + Enter to move backwords through the tab order
                    if (e.Shift)
                    {
                        ProcessTabKey(false);
                    }
                    else
                    {
                        ProcessTabKey(true);
                    }
                }
            }
        }

        private void Form1_Load(object sender, EventArgs e)
        {

            if (mdlGeneral.gblPrliviges > 2)
            {
                Interaction.MsgBox("You are not authorized, illegal operating.", MsgBoxStyle.Information, My.MyProject.Application.Info.ProductName);
                Close();
                return;
            }
            mdlGeneral.GetConnectionString();
            LoadDB();

        }

        private void LoadDB()
        {
            try
            {
                string CommandText = "SELECT * FROM USERS ORDER BY USERID";
                mdlGeneral.ExecuteDatasetQuery(mdlGeneral.SqlDataAdapter, mdlGeneral.SqlDataSet, CommandText, "USERS", mdlGeneral.Connection);
                mdlGeneral.SqlDataAdapter.Fill(UserDataSet, "USERS");
                UserbindingSource.DataSource = UserDataSet;
                UserbindingSource.DataMember = "USERS";
                txtUserID.DataBindings.Add("TEXT", UserbindingSource, "USERID");
                txtUserName.DataBindings.Add("TEXT", UserbindingSource, "USERNAME");
                txtName.DataBindings.Add("TEXT", UserbindingSource, "NAME");
                cmbUserType.DataBindings.Add("TEXT", UserbindingSource, "USERROLE");
                txtLevel.DataBindings.Add("TEXT", UserbindingSource, "SECURITYLEVEL");
                txtPwd.DataBindings.Add("TEXT", UserbindingSource, "PASSWORD");
                txtCNIC.DataBindings.Add("TEXT", UserbindingSource, "CARDNUMBER");
                Addition = false;
                Editing = false;
                ToolStripSaveButton.Enabled = false;
                Panel1.Enabled = false;
                objCellStyle.BackColor = Color.WhiteSmoke;
                UsersDataGridView.AlternatingRowsDefaultCellStyle = objCellStyle;
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message, MsgBoxStyle.Information, My.MyProject.Application.Info.ProductName);
                Close();
            }
        }

        private void BindingNavigatorAddNewItem_Click(object sender, EventArgs e)
        {
            Panel1.Enabled = true;
            Addition = true;
            RollBack = true;
            Editing = false;
            ToolStripSaveButton.Enabled = true;
            ToolStripCloseButton.Text = "&Cancel";
            txtUserName.Focus();
        }

        private void ToolStripEditButton_Click(object sender, EventArgs e)
        {

            EditString = txtUserID.Text.Trim().ToUpper();
            Editing = true;
            Addition = false;
            Panel1.Enabled = true;
            ToolStripSaveButton.Enabled = true;
            ToolStripCloseButton.Text = "&Cancel";
            EditString = txtUserID.Text.Trim();
            txtUserName.Focus();

        }

        private void ToolStripCloseButton_Click(object sender, EventArgs e)
        {
            if (ToolStripCloseButton.Text == "&Close")
            {
                // myConn.Close()
                UserbindingSource.Dispose();
                Close();
            }
            else
            {
                ToolStripCloseButton.Text = "&Close";
                ToolStripSaveButton.Enabled = false;
                Panel1.Enabled = false;
                RefreshDB();
                Addition = false;
                Editing = false;
            }
        }

        private void RefreshDB()
        {
            try
            {
                UserDataSet.Clear();
                string CommandText = "SELECT * FROM USERS ORDER BY USERID";
                mdlGeneral.ExecuteDatasetQuery(mdlGeneral.SqlDataAdapter, mdlGeneral.SqlDataSet, CommandText, "USERS", mdlGeneral.Connection);
                mdlGeneral.SqlDataAdapter.Fill(UserDataSet, "USERS");
                UserbindingSource.DataSource = UserDataSet;
                UserbindingSource.DataMember = "USERS";
                UsersDataGridView.RefreshEdit();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message, MsgBoxStyle.Information, My.MyProject.Application.Info.ProductName);
                Close();
            }
        }

        private void BindingNavigatorDeleteItem_Click(object sender, EventArgs e)
        {

            string DeleteQuery = string.Empty;
            if (ToolStripCloseButton.Text == "&Cancel")
                ToolStripCloseButton.Text = "&Close";
            if (Interaction.MsgBox("Are you sure to delete ('" + txtUserName.Text.Trim().ToUpper() + "') ?", (MsgBoxStyle)((int)MsgBoxStyle.Information + (int)MsgBoxStyle.OkCancel), My.MyProject.Application.Info.ProductName) == MsgBoxResult.Ok)
            {
                DeleteQuery = "DELETE FROM USERS WHERE USERNAME = '" + txtUserName.Text.Trim() + "'";
                mdlGeneral.ExecuteDatasetQuery(mdlGeneral.SqlDataAdapter, mdlGeneral.SqlDataSet, DeleteQuery, "USERS", mdlGeneral.Connection);
            }
            else
            {
                UsersDataGridView.RefreshEdit();
                UserBindingNavigator.Refresh();
                ToolStripCloseButton_Click(sender, e);
                return;
            }
        }

        private void ToolStripSaveButton_Click(object sender, EventArgs e)
        {
            string MaxID = mdlGeneral.GenerateMax_ID("USERS", "USERID").ToString();
            string Edit_Update_Query = string.Empty;
            if (Editing == true & Addition == false)
                Edit_Update_Query = "UPDATE USERS SET USERNAME = '" + txtUserName.Text.Trim().ToLower() + "', PASSWORD = '" + txtPwd.Text.Trim() + "', USERROLE = '" + Strings.StrConv(cmbUserType.Text.Trim(), VbStrConv.ProperCase) + "', SECURITYLEVEL = '" + txtLevel.Text.Trim() + "', NAME = '" + Strings.StrConv(txtName.Text.Trim(), VbStrConv.ProperCase) + "'  WHERE USERID = '" + Conversion.Val(EditString) + "'";
            if (Addition == true & Editing == false)
                Edit_Update_Query = "INSERT INTO USERS VALUES('" + MaxID + "','" + txtUserName.Text.Trim().ToLower() + "','" + txtPwd.Text.Trim() + "','" + Strings.StrConv(txtName.Text.Trim(), VbStrConv.ProperCase) + "','" + Strings.StrConv(cmbUserType.Text.Trim(), VbStrConv.ProperCase) + "','" + Conversion.Val(txtLevel.Text.Trim().ToUpper()) + "',0,'" + txtCNIC.Text.Trim() + "')";
            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, Edit_Update_Query, mdlGeneral.Connection);
            RefreshDB();
            ToolStripSaveButton.Enabled = false;
            Panel1.Enabled = false;
            ToolStripCloseButton.Text = "&Close";
            mdlGeneral.Connection.Close();
        }

        private void txtName_KeyPress(object sender, KeyPressEventArgs e)
        {
            mdlGeneral.ValidateAlphaKeyPress(sender, e, txtName);
        }

        private void txtUserName_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (e.KeyChar == '\b')
                e.Handled = false;
        }

        private void txtPwd_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (e.KeyChar == '\b')
                e.Handled = false;
        }

        private void cmbUserType_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!(char.IsLetter(e.KeyChar) | char.IsControl(e.KeyChar)))
                e.Handled = true;
        }

        private void cmbUserType_SelectedIndexChanged(object sender, EventArgs e)
        {

            if (cmbUserType.Text == "Administrator")
            {
                txtLevel.Text = 1.ToString();
            }
            else if (cmbUserType.Text == "Super User")
            {
                txtLevel.Text = 2.ToString();
            }
            else if (cmbUserType.Text == "User")
            {
                txtLevel.Text = 3.ToString();
            }

        }

    }
}